### ISSUE TITLE
[Minor] Solar Panels show power gain as 1 instead of proper values.

### ISSUE THREAD

**Issue by neronix17** [neronix17/-O21-WikiRim#3]
Check all power systems, might be a miscalculation resulting in incorrect values being displayed, or displaying a multiplier instead of a worthwhile value.

**Comment by schultzcole** [comment]
I did some investigation on this. In the `ThingDef` for `SolarGenerator`, `basePowerConsumption` is set to -1. This is because the code that actually determines the desired power output for a solar generator is hard coded to lerp between 1700 and 0 based on what time of day it is. You can see this for yourself by decompiling Assembly-CSharp.dll and looking at `Rimworld.CompPowerPlantSolar.DesiredPowerOutput`:

``` csharp
protected override float DesiredPowerOutput => Mathf.Lerp(0f, 1700f, parent.Map.skyManager.CurSkyGlow) * RoofedPowerOutputFactor;
```

So the issue isn't a bug in this mod, it's that the power output for solar power plants is coded poorly.

The only solution I can think of is to hard code the power output in the mod as well to match the hard coded value in the game, but obviously that's quite fragile and not optimal.

**Comment by neronix17** [comment]
Yeah figured that out and left it alone for now at least because it seemed to be too specific a case to deal with easily, only real solution I can think of would be to have it say "Variable Output" instead of the number.
